ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUpdaterThread.java

```diff
@@ -94,6 +98,8 @@
   private BlockingQueue<AnalyzeWork> workQueue;
   private Thread[] workers;
 
+  private Set<String> dbsBeingFailedOver;
```

Comment:
Why do we need it at instance level?

Reply:
This is the only way we can access this set within each iteration of StatsUpdater and also within each execution of actual analysis work after dequeuing from the worker queue. This set is clear when new iteration of this thread kicks in.